### ISSUE TITLE
Fix DBUS interface for recent KDE Connect versions

### ISSUE THREAD

**Issue by pdf** [pdf/kdeconnect-chrome-extension#56]
Looks like KDE Connect have removed the old `*Trusted` properties/signals, and now uses `*PairState` need to update the DBUS interfaces and UI logic to use the new properties if they exist.

**Comment by albertvaka** [comment]
Sorry for breaking this. "isTrusted" is called "isPaired" now.

**Comment by pdf** [comment]
Yeah, I have the DBUS side sorted, just need to plumb it through the NativeHost interface to the UI, and have ended up down a rabbit hole of extension manifest v3, serviceworkers and general refactoring.

**Comment by hemanti0503** [comment]
Hey @pdf, what's the status on this? I too see my devices in Firefox, but also the red exclamation mark and can't send, unfortunately.

**Comment by pdf** [comment]
Sorry for the delay, IRL stuff has had me out of town and time-poor the last couple of weeks.

**Comment by hemanti0503** [comment]
Thanks for the update! It's great seeing you still working on this. Thank you for all your work!

**Comment by sudoshindo** [comment]
Subscribing to this issue, waiting for fix.

**Comment by carnager** [comment]
until (if) this gets fixed, one can use the external application addon, which lets you chose an external binary with arguments and custom icon.

**Comment by sudoshindo** [comment]
> until (if) this gets fixed, one can use the external application addon, which lets you chose an external binary with arguments and custom icon.

Where is this located

**Comment by carnager** [comment]
I don't know if external links are ok here... but in worst case it gets deleted:
https://addons.mozilla.org/de/firefox/addon/external-application/
https://chromewebstore.google.com/detail/external-application-laun/bifmfjgpgndemajpeeoiopbeilbaifdo?pli=1

### Steps to make this work:

1. Get your device ID: `kdeconnect-cli -l`
2. Open Extensions Preferences
3. Fill in title
4. Fill in path to kdeconnect-cli
5. for arguments use: `-d ID --share [HREF]`
6. Enable toolbar Icon option and choose Page, Link, Image and Tab (or a subset of your desire)
7. Choose an Icon
8. Select "Surround Arguments with Quote characters"
9. Hit "Add Application"

Now you should have a new icon in your extensions button menu. Once you click this it opens a page with the link for the native application. Download this, extract it, run ./install.sh

### Multiple Devices

You don't get a menu with all your devices. For each device you need to add a separate "Application". Fill in the same details and just change the ID in arguments to match your other devices. Also change the title of the Application.
When you rightclick the extension icon you can then  switch your "Applications"

If this seems cumbersome to you, don't call kdeconnect-cli directly, but use your own custom script, which lists all devices using dmenu/rofi/etc and then sends the url to it.

Here is a custom script that uses rofi. the only argument this script needs is `[HREF]`

```
#!/bin/bash

_files=( "$@" )

_id=$(kdeconnect-cli -a --id-name-only 2>&1 | rofi -dmenu -p '> ' -mesg "Choose Device")
if [[ -z "${_id}" ]]; then
    exit
else
    _id=$(awk '{ print $1 }' <<< "${_id}")
fi

for x in "${_files[@]}"; do
    kdeconnect-cli -d "${_id}" --share "${x}"
done
```

![Screenshot-2024-02-11-105030](https://github.com/pdf/kdeconnect-chrome-extension/assets/3321178/93d1f036-1e32-4fe0-b47c-0318afbbe017)

Using this approach has another advantage: You can actually see if the device is available. calling kdeconnect-cli directly you will never know if the device was ready to receive data.

**Comment by sudoshindo** [comment]
> firefox

💀

**Comment by carnager** [comment]
> > firefox
> 
> 💀

Sorry, added chrome version too

**Comment by carnager** [comment]
Ok, added Instructions to make this work.

**Comment by sudoshindo** [comment]
@carnager Got it all setup, thanks.

One important thing to note is that the install script provided by the extension creates folders to put the native messaging hosts in even though the corresponding browser isn't installed.

**Comment by zunjack** [comment]
@pdf

**Comment by sudoshindo** [comment]
Someone please maintain this repo.
